payloads/v8/channel.ts

```diff
@@ -32,35 +32,90 @@ export interface APIPartialChannel {
 }
 
 /**
- * https://discord.com/developers/docs/resources/channel#channel-object-channel-structure
+ * This interface is used to allow easy extension for other channel types. While
+ * also allowing `APIPartialChannel` to be used without breaking.
  */
-export interface APIChannel extends APIPartialChannel {
+export interface APIChannelBase<T extends ChannelType> extends APIPartialChannel {
+	type: T;
+}
+
+export type TextChannelTypes = ChannelType.DM | ChannelType.GroupDM | ChannelType.GuildNews | ChannelType.GuildText;
+
+export type GuildChannelTypes = Exclude<
+	TextChannelTypes | ChannelType.GuildVoice | ChannelType.GuildStageVoice,
+	ChannelType.DM | ChannelType.GroupDM
+>;
+
+export interface APITextBasedChannel<T extends ChannelType> extends APIChannelBase<T> {
+	/**
+	 * The id of the last message sent in this channel (may not point to an existing or valid message)
+	 */
+	last_message_id?: Snowflake | null;
+}
+
+export interface APIGuildChannel<T extends ChannelType> extends APIChannelBase<T> {
 	/**
 	 * The id of the guild (may be missing for some channel objects received over gateway guild dispatches)
 	 */
 	guild_id?: Snowflake;
+	/**
+	 * Explicit permission overwrites for members and roles
+	 *
+	 * See https://discord.com/developers/docs/resources/channel#overwrite-object
+	 */
+	permission_overwrites?: APIOverwrite[];
 	/**
 	 * Sorting position of the channel
 	 */
 	position?: number;
 	/**
-	 * Explicit permission overwrites for members and roles
+	 * ID of the parent category for a channel (each parent category can contain up to 50 channels)
 	 *
-	 * See https://discord.com/developers/docs/resources/channel#overwrite-object
+	 * OR
+	 *
+	 * ID of the parent channel for a thread
 	 */
-	permission_overwrites?: APIOverwrite[];
+	parent_id?: Snowflake | null;
+	/**
+	 * Whether the channel is nsfw
+	 */
+	nsfw?: boolean;
+}
+
+export type GuildTextChannelTypes = Exclude<TextChannelTypes, ChannelType.DM | ChannelType.GroupDM>;
```

Comment:
This won't be needed soon 🎉

Reply:
Should this be fixed now, or will we wait for the upstream PR?

Reply:
Wait, this will be needed still, n e v e r m i n d  m e

Reply:
See edit, I'm apparently all over the place